libimage/manifests/manifests.go

```diff
@@ -236,6 +245,87 @@ func (l *list) Reference(store storage.Store, multiple cp.ImageListSelection, in
 			}
 		}
 	}
+	if len(l.artifacts.Manifests) > 0 {
+		img, err := is.Transport.GetImage(s)
+		if err != nil {
+			return nil, fmt.Errorf("locating image %s: %w", transports.ImageName(s), err)
+		}
+		imgDirectory, err := store.ImageDirectory(img.ID)
+		if err != nil {
+			return nil, fmt.Errorf("locating per-image directory for %s: %w", img.ID, err)
+		}
+		tmp, err := os.MkdirTemp(imgDirectory, "referenced-artifacts")
```

Comment:
Magic string, should this be a constant?

Reply:
Eh, maybe?  Sure, changing it.